src/execution/join_hashtable.cpp

```diff
@@ -86,32 +120,196 @@ void JoinHashTable::Merge(JoinHashTable &other) {
+		for (idx_t i = 0; i < remaining_count; i++) {
+			const auto row_index = remaining_sel->get_index(i);
+
+			idx_t &ht_offset = ht_offsets[row_index];
+			bool occupied;
+			aggr_ht_entry_t entry;
+
+			if (USE_SALTS) {
+
+				hash_t row_salt = salts[row_index];
+
+				// increment the ht_offset of the entry as long as next entry is occupied and salt does not match
+				while (true) {
+					entry = entries[ht_offset];
+					occupied = entry.IsOccupied();
+					bool salt_match = entry.GetSalt() == row_salt;
+
+					// condition for incrementing the ht_offset: occupied and row_salt does not match -> move to next
+					// entry
+					if (!occupied || salt_match) {
+						break;
+					}
+
+					IncrementAndWrap(ht_offset, ht->bitmask);
+				}
+			} else {
+				entry = entries[ht_offset];
+				occupied = entry.IsOccupied();
+			}
+
+			// the entries we need to process in the next iteration are the ones that are occupied and the row_salt
+			// does not match, the ones that are empty need no further processing
+			state.salt_match_sel.set_index(salt_match_count, row_index);
+			salt_match_count += occupied;
+
+			// entry might be empty, so the pointer in the entry is nullptr, but this does not matter as the row
+			// will not be compared anyway as with an empty entry we are already done
+			row_ptr_insert_to[row_index] = entry.GetPointerOrNull();
+		}
+
+		if (salt_match_count != 0) {
+
+			// Perform row comparisons, after function call salt_match_sel will point to the keys that match
+
+			idx_t key_match_count = ht->row_matcher_build.Match(keys, key_state.vector_data, state.salt_match_sel,
+			                                                    salt_match_count, ht->layout, state.row_ptr_insert_to_v,
+			                                                    &state.key_no_match_sel, key_no_match_count);
+
+			D_ASSERT(key_match_count + key_no_match_count == salt_match_count);
+
+			// Set a pointer to the matching row
+			for (idx_t i = 0; i < key_match_count; i++) {
+				const auto row_index = state.salt_match_sel.get_index(i);
+				pointers_result[row_index] = row_ptr_insert_to[row_index];
+
+				match_sel.set_index(match_count, row_index);
+				match_count++;
+			}
+
+			// Linear probing: each of the entries that do not match move to the next entry in the HT
+			for (idx_t i = 0; i < key_no_match_count; i++) {
+				const auto row_index = state.key_no_match_sel.get_index(i);
+				auto &ht_offset = ht_offsets[row_index];
+
+				IncrementAndWrap(ht_offset, ht->bitmask);
+			}
+		}
+
+		remaining_sel = &state.key_no_match_sel;
+		remaining_count = key_no_match_count;
+	}
+}
+
+inline bool JoinHashTable::UseSalt() const {
+	// only compare salts with the ht entries if the capacity is larger than 8192 so
+	// that it does not fit into the CPU cache and if there is only one equality condition as otherwise
+	// we potentially need to compare multiple keys
+	return this->capacity > 8192 && this->equality_predicate_columns.size() == 1;
```

Comment:
Could the `8192` be a `static constexpr const idx_t USE_SALT_THRESHOLD = 8192;` defined the header with this descriptive comment there instead?